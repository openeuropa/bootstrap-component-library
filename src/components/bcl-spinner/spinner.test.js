import { renderTwigFileAsNode } from "@openeuropa/bcl-test-utils";
import demoData from "@openeuropa/bcl-data-spinner/data";

const template = "@oe-bcl/bcl-spinner/spinner.html.twig";
const render = (params) => renderTwigFileAsNode(template, params, true);

describe("OE - Spinner", () => {
  test(`renders correctly`, () => {
    expect.assertions(1);

    return expect(render(demoData)).resolves.toMatchSnapshot();
  });

  test(`renders small correctly`, () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, small: true })
    ).resolves.toMatchSnapshot();
  });

  test(`renders grow correctly`, () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, type: "grow" })
    ).resolves.toMatchSnapshot();
  });
});
