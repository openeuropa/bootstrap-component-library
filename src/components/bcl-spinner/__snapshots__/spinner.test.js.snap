// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OE - Spinner renders correctly 1`] = `
<jest>
  <div
    class="spinner-border text-danger"
    role="status"
  >
    <span
      class="visually-hidden"
    >
      Loading...
    </span>
  </div>
</jest>
`;

exports[`OE - Spinner renders grow correctly 1`] = `
<jest>
  <div
    class="spinner-grow text-danger"
    role="status"
  >
    <span
      class="visually-hidden"
    >
      Loading...
    </span>
  </div>
</jest>
`;

exports[`OE - Spinner renders small correctly 1`] = `
<jest>
  <div
    class="spinner-border spinner-border-sm text-danger"
    role="status"
  >
    <span
      class="visually-hidden"
    >
      Loading...
    </span>
  </div>
</jest>
`;
